cdap-api/src/main/java/co/cask/cdap/api/security/securestore/SecureStoreMetadata.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.security.securestore;
+
+import java.util.Date;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents the metadata for the data stored in the Secure Store.
+ */
+public final class SecureStoreMetadata {
+
+  private static final String DESCRIPTION_FIELD = "description";
+  private static final String DESCRIPTION_DEFAULT = "";
+
+  private final String name;
+  private final String description;
+  private final long createdEpochMs;
+  private final Map<String, String> properties;
+
+  private SecureStoreMetadata(String name, String description, Date created, Map<String, String> properties) {
+    this.name = name;
+    this.description = description;
+    this.createdEpochMs = created.getTime();
+    this.properties = properties;
+  }
+
+  public static SecureStoreMetadata of(String name, Map<String, String> properties) {
+    String tempDescription = properties.get(DESCRIPTION_FIELD) == null ?
```

Comment:
actually, why do we need to read description from properties? If users want to provide description, perhaps have a constructor that accepts a description?

So,
```
public SecureStoreMetadata(String name, @Nullable String description, Map<String, String> properties) {
  this.description = description == null ? DESCRIPTION_DEFAULT : description;
  this.date = new Date();
  ...
}

public SecureStoreMetadata(String name, Map<String, String> properties) {
  this(name, null, properties);
}
```